src/utilities/device_utils.c

```diff
@@ -526,6 +526,61 @@ hypre_dim3(HYPRE_Int x, HYPRE_Int y, HYPRE_Int z)
    return d;
 }
 
+/*--------------------------------------------------------------------------
+ * hypreGPUKernel_ArrayToArrayOfPtrs
+ *--------------------------------------------------------------------------*/
+
+template <typename T>
+__global__ void
+hypreGPUKernel_ArrayToArrayOfPtrs( hypre_DeviceItem  &item,
+                                   HYPRE_Int          n,
+                                   HYPRE_Int          m,
+                                   T                 *data,
+                                   T                **data_aop )
+{
+   HYPRE_Int i = hypre_gpu_get_grid_thread_id<1, 1>(item);
+
+   if (i < n)
+   {
+      data_aop[i] = &data[i * m];
+   }
+}
+
+/*--------------------------------------------------------------------
+ * hypreDevice_ArrayToArrayOfPtrs
+ *--------------------------------------------------------------------*/
+
+template <typename T>
+HYPRE_Int
+hypreDevice_ArrayToArrayOfPtrs(HYPRE_Int n, HYPRE_Int m, T *data, T **data_aop)
+{
+   /* Trivial case */
+   if (n <= 0)
+   {
+      return hypre_error_flag;
+   }
+
+   dim3 bDim = hypre_GetDefaultDeviceBlockDimension();
+   dim3 gDim = hypre_GetDefaultDeviceGridDimension(n, "thread", bDim);
+
+   HYPRE_GPU_LAUNCH( hypreGPUKernel_ArrayToArrayOfPtrs, gDim, bDim, n, m, data, data_aop);
+
+   return hypre_error_flag;
+}
+
+/*--------------------------------------------------------------------
+ * hypreDevice_ComplexArrayToArrayOfPtrs
+ *--------------------------------------------------------------------*/
+
+HYPRE_Int
+hypreDevice_ComplexArrayToArrayOfPtrs(HYPRE_Int       n,
```

Comment:
The name of this function is a bit confusing. I understood what it is after reading the code. So, it basically just gets the addresses of the elements of `data` and save them in another array.

Reply:
Something like "GetArrayElemPtrs"? The current one reads like we are converting `Complex` to `Complex *`. Thanks!

Reply:
Sounds good to me. Thank you!